src/Chip/index.tsx

```diff
@@ -2,73 +2,79 @@ import React, { ForwardedRef, KeyboardEventHandler, MouseEventHandler } from 're
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
 // @ts-ignore
-import Icon from '../Icon';
+import { requiredWhen } from '../utils/propTypes';
+// @ts-ignore
+import { STYLE_VARIANTS } from './constants';
+// @ts-ignore
+import ChipIcon from './ChipIcon';
 
-const STYLE_VARIANTS = [
-  'light',
-  'dark',
-];
+export const CHIP_PGN_CLASS = 'pgn__chip';
 
 export interface IChip {
   children: React.ReactNode,
   className?: string,
   variant?: string,
   iconBefore?: React.ReactElement | Function,
+  iconBeforeAlt?: string,
   iconAfter?: React.ReactElement | Function,
+  iconAfterAlt?: string,
   onIconBeforeClick?: KeyboardEventHandler & MouseEventHandler,
   onIconAfterClick?: KeyboardEventHandler & MouseEventHandler,
   disabled?: boolean,
+  isSelected?: boolean,
 }
 
-export const CHIP_PGN_CLASS = 'pgn__chip';
-
 const Chip = React.forwardRef(({
   children,
   className,
   variant,
   iconBefore,
+  iconBeforeAlt,
   iconAfter,
+  iconAfterAlt,
   onIconBeforeClick,
   onIconAfterClick,
   disabled,
+  isSelected,
   ...props
 }: IChip, ref: ForwardedRef<HTMLDivElement>) => (
   <div
+    tabIndex={0}
+    role="button"
```

Comment:
If the left/right icons included in the `Chip` are interactive, I believe the intent is to _not_ have the top-level `Chip` be interactive as a button, where _only_ the icon(s) are interactive. This prevents possible usability issues where user is trying to dismiss the `Chip` but accidentally clicks on the parent `Chip` container instead of the `IconButton`.

Similarly, if the icon(s) are interactive, the parent `Chip` should remain unchanged from a hover styles perspective. The only hover focus in this case should be on the `IconButton` itself.

Reply:
Great points! I've refactored some stuff here and updated docs and I believe I achieved the behaviour you described, so please take a look again and let me know what you think 🙂